stdlib/REPL/src/REPL.jl

```diff
@@ -436,9 +436,15 @@ mutable struct LineEditREPL <: AbstractREPL
     last_shown_line_infos::Vector{Tuple{String,Int}}
     interface::ModalInterface
     backendref::REPLBackendRef
-    LineEditREPL(t,hascolor,prompt_color,input_color,answer_color,shell_color,help_color,history_file,in_shell,in_help,envcolors) =
+    function LineEditREPL(t,hascolor,prompt_color,input_color,answer_color,shell_color,help_color,history_file,in_shell,in_help,envcolors)
+        opts = Options()
+        opts.hascolor = hascolor
+        if !hascolor
+            opts.beep_colors = [""]
+        end
```

Comment:
Maybe this logic belongs in the `Options` constructor taking kwargs? e.g. in the body of that method, replace `beep_colors` by `hascolor ? beep_colors : [""]` (or change the default value by `hascolor ? ["\e[90m"] : [""]`).